### PR TITLE
Use Metal on macOS

### PR DESCRIPTION
**Pull request by reyamir**

### CODE REVIEW

File: crates/renderer/src/skia/metal_surface.rs
Comment (marc2332) [inline crates/renderer/src/skia/metal_surface.rs:1]: Where is the surface? 🤔

File: crates/renderer/src/skia/gl_surface.rs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        GlSurface, Surface as GlutinSurface, SurfaceAttributesBuilder, SwapInterval, WindowSurface,
+    },
+};
+use std::{ffi::CString, num::NonZeroU32};
+use winit::{raw_window_handle::HasWindowHandle, window::Window};
+use winit::dpi::PhysicalSize;
+
+use freya_engine::prelude::*;
+
+pub struct SkiaSurface {
```
Comment (marc2332) [inline crates/renderer/src/skia/gl_surface.rs:20]: I would probably rename this to something like GlSurface and then have a type alias be stored in the window state instead, same thing for the metal surface `MetalSurface`

File: crates/renderer/src/skia/metal_surface.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+use objc::runtime::YES;
+use winit::{raw_window_handle::HasWindowHandle, window::Window};
+
+use freya_engine::prelude::*;
+
+pub struct SkiaSurface {
+    pub(crate) gr_context: DirectContext,
+    pub(crate) metal_layer: MetalLayer,
+    pub(crate) command_queue: CommandQueue,
+    pub(crate) scale_factor: f64,
```
Comment (marc2332) [inline crates/renderer/src/skia/metal_surface.rs:14]: I don't think scale_factor should stored in the surface

**Review by marc2332 — CHANGES_REQUESTED** [review]

**Comment by reyamir** [comment]
I will rework on this on this weekend, as well as add vulkan support too.

**Comment by marc2332** [comment]
hey @reyamir ! I just opened https://github.com/marc2332/freya/pull/822 which should make adding metal easier :)

**Comment by reyamir** [comment]
> hey @reyamir ! I just opened #822 which should make adding metal and other drivers easier :)

that great, I will rewrite my PR

**Pull request closed without merging**